src/Doctrine/DoctrineHelper.php

```diff
@@ -122,6 +124,43 @@ public function isClassAnnotated(string $className): bool
         return false;
     }
 
+    public function doesClassUsesAttributes(string $className): bool
+    {
+        /** @var EntityManagerInterface $em */
+        $em = $this->getRegistry()->getManagerForClass($className);
+
+        if (null === $em) {
+            throw new \InvalidArgumentException(sprintf('Cannot find the entity manager for class "%s"', $className));
+        }
+
+        if (null === $this->annotatedPrefixes) {
+            // doctrine-bundle <= 2.2
+            $metadataDriver = $em->getConfiguration()->getMetadataDriverImpl();
+
+            if (!$this->isInstanceOf($metadataDriver, MappingDriverChain::class)) {
+                return $metadataDriver instanceof AttributeDriver;
+            }
+
+            foreach ($metadataDriver->getDrivers() as $namespace => $driver) {
+                if (0 === strpos($className, $namespace)) {
+                    return $driver instanceof AttributeDriver;
+                }
+            }
+
+            return $metadataDriver->getDefaultDriver() instanceof AttributeDriver;
+        }
+
+        $managerName = array_search($em, $this->getRegistry()->getManagers(), true);
+
+        foreach ($this->annotatedPrefixes[$managerName] as [$prefix, $attributeDriver]) {
+            if (0 === strpos($className, $prefix)) {
+                return null !== $attributeDriver;
+            }
+        }
+
+        return false;
+    }
```

Comment:
Could we share code from `isClassAnnotated()`? Perhaps we create a new:

```php
private function doesClassUseDriver(string $className, string $driverClass): bool
```

which contains all the logic, but where `$driverClass` is used instead of `AnnotationDriver` or `AttributeDriver`. Then, this method would be:

```php
public function doesClassUseAttributes(string $className): bool
{
    return $this->doesClassUseDriver(string $className, AttributeDriver::class): bool
}
```